### PR TITLE
Update specs to support visionOS

### PR DESCRIPTION
**Pull request by gabrieldonadel**
Update specs for https://github.com/CocoaPods/CocoaPods/pull/11965

Related to https://github.com/CocoaPods/CocoaPods/issues/11961

### CODE REVIEW

File: install_header_mappings_dir/after/Pods/Manifest.lock [hunk 1]
```diff
@@ -14,5 +14,5 @@ EXTERNAL SOURCES:
     :path: HeaderMappingsDirPod
 
 SPEC CHECKSUMS:
-  DepPod: 5f07a9509c00db5f4a674e3d56baea1724f8ea3b
-  HeaderMappingsDirPod: 6657416e95eb86734ddbf92abf9fcab8610551b9
+  DepPod: 99ce4035871faac7ded80730803c36226db7d304
```
Comment (dnkoutso) [inline install_header_mappings_dir/after/Pods/Manifest.lock:17]: This seems suspicious to have changed? Why would the checksums change

Comment (gabrieldonadel) [reply install_header_mappings_dir/after/Pods/Manifest.lock:17]: Not really sure to be honest, I tried rerunning `bundle exec rake spec:rebuild_integration_fixtures` in a clean branch after running `git clean -dfx .` and this was changes would still show up

Comment (gabrieldonadel) [reply install_header_mappings_dir/after/Pods/Manifest.lock:17]: But I would guess that's due to the [podspec.json](https://github.com/CocoaPods/cocoapods-integration-specs/pull/343/files/145d5005fdafbc178e10b773a711ef6a0f0a5700#diff-aa3d0529fd532876bd8fac0d07a21d1a55b352870509959a9d1c3a78d57ffdda) change adding `"visionos": null,`

Comment (dnkoutso) [reply install_header_mappings_dir/after/Pods/Manifest.lock:17]: that makes sense as to why it changed...............this is a bit weird? I do not recall CocoaPods encoding this info this means if someone bumps to the latest version all of these checksums would change, maybe that was always the case. We can see if tvOS did the same or iPadOS.

Comment (gabrieldonadel) [reply install_header_mappings_dir/after/Pods/Manifest.lock:17]: @dnkoutso couldn't find any commits that old on this repo, seems that most of these Podfile.locks never had any platform-related changes?

Comment (dnkoutso) [reply install_header_mappings_dir/after/Pods/Manifest.lock:17]: @gabrieldonadel spoke with the team a bit and I think we will proceed with this as it is by design.

See old PRs:

https://github.com/CocoaPods/Core/pull/279 and here https://github.com/CocoaPods/Core/pull/282

**Review by dnkoutso — APPROVED** [review]

**Comment by dnkoutso** [comment]
@gabrieldonadel there is no CI for this repo. I merged it. Please bump your original CocoaPods/CocoaPods PR SHA of this submodule (usually to latest master) and push for your PR to get green.

**Pull request merged**